src/payment.js

```diff
@@ -0,0 +1,174 @@
+      }),
+    )
+    .then((paypalCheckoutInstance) => {
+      const paramsPayment = params.payment || {};
+      const payment = {
+        flow: 'vault',
+        currency: paramsPayment.currency || 'USD',
+        amount: paramsPayment.amount || order.grand_total,
+      };
+      return paypal
+        .Buttons({
+          style: params.style || {},
+          createBillingAgreement: () => paypalCheckoutInstance.createPayment(payment),
+          onApprove: (data, actions) =>
+            paypalCheckoutInstance
+              .tokenizePayment(data)
+              .then((card) => methods(request).create(order.id, payment.amount, 'paypal', card))
+              .then((result) => (isFunction(params.onDone) ? params.onDone(result) : result)),
+          onCancel: isFunction(params.onCancel)
+            ? () => params.onCancel()
+            : () => console.log('PayPal payment cancelled'),
+          onError: isFunction(params.onError)
+            ? (err) => params.onError(err)
+            : (err) => console.error('PayPal error', err),
+        })
+        .render(params.elementId || '#paypal-button');
+    })
+    .catch(
+      isFunction(params.onError) ? params.onError : (err) => console.error('PayPal error', err),
+    );
+}
+
+async function stripeElements(request, params, gateways) {
+  const submitButton = document.getElementById(params.submitButtonId || 'stripe-submit-button');
+  if (!submitButton) {
+    throw new Error('Submit button not found');
+  }
+  const { public_key: publicKey } = gateways.stripe;
+  const stripe = window.Stripe(publicKey);
+  const elements = stripe.elements();
+  let card = null;
+  const createElement = (type) => {
+    const elementParams = params[type] || {};
+    const elementOptions = elementParams.options || {};
+    const element = elements.create(type, elementOptions);
+    element.mount(elementParams.elementId || `#${type}-element`);
+    if (type === 'card' || type === 'cardNumber') {
+      card = element;
+    }
+  };
+
+  if (params.separateElements) {
+    createElement('cardNumber');
+    createElement('cardExpiry');
+    createElement('cardCvc');
+  } else {
+    createElement('card');
+  }
+
+  const onSubmit = async () => {
+    const onError = (error) => {
+      if (isFunction(params.onError)) {
+        return params.onError(error);
+      }
+      throw new Error(error.message);
+    };
+
+    const token = await stripe
+      .createToken(card)
+      .then(({ token, error }) => (error ? onError(error) : token));
+
+    await cardApi
+      .createToken({
+        nonce: token.id,
+        last4: token.card.last4,
+        exp_month: token.card.exp_month,
+        exp_year: token.card.exp_year,
+        brand: token.card.brand,
+        address_check: token.card.address_line1_check,
+        cvc_check: token.card.cvc_check,
+        zip_check: token.card.address_zip_check,
+      })
+      .then((result) => (isFunction(params.onDone) ? params.onDone(result) : result));
```

Comment:
@vstudenichnik-insoft similar to paypal, it should call to update cart here

Reply:
Done